.github/contributing.md

```diff
@@ -0,0 +1,245 @@
+
+  - Only avail of docblock syntax to further document types when language doesn't allow concise type declarations.
+
+
+<br>
+
+
+## PR creation and management
+
+To add code to our codebase after following all above steps, you *must* create a new PR for your
+changes.
+
+The destination branch normally is `dev`, ensure this is the case.
+
+Follow and fill in the `PULL_REQUEST_TEMPLATE` present.<br>
+Link corresponding PR to the issue at hand by adding `Closes #{ISSUE-ID}` somewhere in `Description, Changes` section to PR body or add it manually.
+
+> NOTE: Manual option is only available after PR creation from `Development` sidebar option.
+
+If unsure about whether PR is ready for review, create a draft PR.<br>
+Update PR to be full PR and ask for reviewers when ready.
+
+If changes are requested,
+1. Ask for clarification on changes necessary, if needed.
+2. Don't be afraid to state your point of view on proposed changes.
+3. Debate respectfully, acknowledge the other contributors point of view
+4. Finalize decision and solution to identify whether changes are actually necessary
+- If yes, make changes locally. <br>Follow <span style="color: #5C9340;">## Pre-committing to Github</span> guidelines defined above and push changes to Github
+5. Request re-review from contributors once ready again.
+
+> NOTE: Every PR __must__ require at least 2 approvals before merge-ready.
+
+
+<br>
+
+If changes elsewhere are pushed to the `dev` branch, your branch will need to pick up these new changes before adding your changes on top.
+
+This inevitably means any branches that are at PR or review stage will need to be rebased with `dev` before merging
+
+1. Navigate to the local branch you are completing the changes upon.
+2. Execute `git rebase dev`.
+3. Observe potential merge conflicts
+- If yes, resolve merge conflicts and execute `git rebase --continue` to resume the merge of `dev` new code into your branch
+- If no, message indicates `Successfully rebased with dev` in CLI
+
+<br>
+
+
+## PR Review process
+
+Upon request, a contributor is obligate to review another contributor's PR.<br>
+The following guidelines need to be followed upon undertaking a review.
+
+ Review should be:
+ - Thorough
+ - Articulate
+ - Respectful
+ - Split in two parts:
+
+    Single Line Review Comments should consider:
+    1. Syntax changes needed
+    2. Misunderstanding of code provided
+    3. Formatting adjustments required
+    4. Additions/suggestions for improvements in code
+
+    Overall Review Comment should consider:
+    1. Switch to branch and execute flow targeted by PR
+    2. Observe physical breakages/issues - Report as comment
+    3. Execute all tests surrounding the PR
+    4. Observe failures - Report as comment
+    5. Any edge cases contributor didn't consider - Report in comment
+    6. Overall review sentiment for the contributors
+    7. Next steps
+
+
+ - a structured format for a potential debate
+
+ {Marco and Dario input here}
```

Comment:
not sure what you're looking for, here

Reply:
So, you and Dario most often have been the reviewers. 

Thus, I'm looking to identify some methods/practices/steps ye take as part of your review process and include them here as general guidelines to follow for any following reviewer.

Reply:
EDIT: Since a thorough review process is what's required and yourself and Dario are actually very experienced in this area, that's why I was asking you to provide your input with guidelines or the process you go through when undertaking one.

Reply:
ah, I see! Thanks for clarifying.

This really depends a lot on the context:
- CI health (how much is tested/checked in CI)
- scope of PR
- features of the language (e.g. Haskell vs PHP reviews are quite different)

Anyway, I would just probably link to some more authoritative resources than myself: for example, see https://google.github.io/eng-practices/review/reviewer/ or https://chelseatroy.com/2019/12/18/reviewing-pull-requests/

Reply:
Maybe we could add (even though it's not a thing yet) that the test coverage should never go down with a PR?

Reply:
> Maybe we could add (even though it's not a thing yet) that the test coverage should never go down with a PR?

If we want such a rule, we should check it in CI.

Still, I'm not sure it works as a strict rule. I'd prefer to have a trend and check the overall behaviour, not the local one